libs/types/src/smartcard_auth/dipped_smartcard_auth.ts

```diff
@@ -30,17 +30,19 @@ export interface RemoveCard {
   readonly user: SystemAdministratorUser | ElectionManagerUser;
 }
 
+export type CardSummaryNotReadyStatus = 'no_card' | 'error';
+
 export interface SystemAdministratorLoggedIn {
   readonly status: 'logged_in';
   readonly user: SystemAdministratorUser;
-  readonly card?: CardStorage & CardProgramming;
+  readonly card: CardSummaryNotReadyStatus | (CardStorage & CardProgramming);
```

Comment:
I'm a little wary of changing the type of `card` for dipped auth but not inserted auth. I've been trying to keep them matching to reduce cognitive overhead. Maybe instead we can add an intersecting card status field and change all the `card` types in all the auth hooks?

e.g.

```ts
// For user types with programming and storage
readonly card: { status: 'no_card' | 'error' } | { status: 'ready' } & CardStorage & CardProgramming
// For user types with just storage
readonly card: { status: 'no_card' | 'error' } | { status: 'ready' } & CardStorage
```

or if we want to get fancy and make it more concise:

```ts
type Card<TCardActions> = { status: 'no_card' | 'error' } | { status: 'ready' } & TCardActions
// For user types with programming and storage
readonly card: Card<CardStorage & CardProgramming>
// For user types with just storage
readonly card: Card<CardStorage>
```

Reply:
This doesn't make sense to me yet. For `InsertedAuth`, why would it be possible to have status be `no_card`?

Right now it makes sense to me that card has a different meaning in each of these cases. My instinct is to:
- Leave `card` for InsertedSmartCard the same as it is
- Remove `card` for DippedSmartCard for `ElectionManagerAuth`, unless we have plans to do card programming or reading in Election Manager mode
- Rename `card` for DippedSmartCard for `SystemAdministratorAuth` to make clearer that it is not the card that was used to log in but is a different card. This was confusing me at first.

Reply:
I am thinking I will still take your suggestion on cleaning up the type to avoid a union of a string and an object, but that type would only be in `Dipped`. 

I don't have ideas for renaming. `cardProgramming` works for me, especially because I don't think `CardStorage` is used anywhere in the apps where we use `Dipped` so that could be removed so simplify the type further. 

Defer to @arsalansufi in case there's more context or more coming down the pipe.

Reply:
I took out the `card` for Election Manager, but didn't pare down the one for Sys Admin because it looks like we're using CardStorage if only for testing and it does make sense to check long and short values have been cleared.

Reply:
Ah yes good catch!